libs/cx/src/lib/side-navigation/side-navigation.component.ts

```diff
@@ -0,0 +1,146 @@
+    /**
+     * Side navigation configuration, to pass whole model object, instead of creating HTML from scratch
+     */
+    @Input()
+    sideNavigationConfiguration: SideNavigationModel;
+
+    /** Whether condensed mode is included */
+    @Input()
+    @HostBinding('class.fdx-side-nav--condensed')
+    condensed = false;
+
+    /** Prevents the side navigation from truncating or wrapping, extending the width to its longest label. */
+    @Input()
+    dynamicWidth = false;
+
+    /**
+     * The screen width, in pixels, at which to automatically collapse the side navigation on window resize.
+     */
+    @Input()
+    collapseWidth: number;
+
+    /**
+     * Whether this side nav should display as narrow. Note this can only be used with side nav items that have icons.
+     */
+    @Input()
+    narrow = false;
+
+    /** Whether clicking on elements should change selected state of items */
+    @Input()
+    set selectable(selectable: boolean) {
+        this.nestedListState.selectable = selectable;
+    }
+
+    /** @hidden */
+    @ContentChild(SideNavigationUtilityDirective)
+    sideNavUtility: SideNavigationUtilityDirective;
+
+    /** @hidden */
+    @ContentChild(SideNavigationMainComponent, { descendants: true })
+    sideNavMain: SideNavigationMainComponent;
+
+    /** @hidden */
+    @ViewChildren(PreparedNestedListComponent)
+    preparedNestedList: QueryList<PreparedNestedListComponent>;
+
+    /** @hidden */
+    constructor(private keyboardService: NestedListKeyboardService, private nestedListState: NestedListStateService) {
+        this.keyboardService.refresh$.subscribe(() => {
+            /** Refresh list of elements, that are being supported by keyboard */
+            this.keyboardService.refreshItems(this.getLists());
+        });
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        /** Set up condensed state */
+        this.nestedListState.condensed =
+            this.condensed || !!(this.sideNavigationConfiguration && this.sideNavigationConfiguration.condensed);
+
+        if (this.collapseWidth) {
+            this.onResize();
+        }
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        if (!this.sideNavigationConfiguration) {
+            this.keyboardService.refreshItems(this.getLists());
+        }
+        if (this.sideNavMain && this.narrow) {
+            this.sideNavMain._setupScrollButtons();
+            this.sideNavMain.list.nestedItems.forEach((item) => {
+                item._narrow = true;
+            });
+        }
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this.sideNavigationConfiguration) {
+            this.keyboardService.refreshItems(this.getLists());
+        }
+    }
+
+    /** @hidden */
+    @HostListener('window:resize')
+    onResize(): void {
+        if (this.collapseWidth) {
+            this.condensed = window.innerWidth <= this.collapseWidth;
```

Comment:
Do not call `window` directly. Use abstraction of Window

Reply:
We will address this app-wide in another PR